File: leetcode/add_one_row_to_tree/tree_new_row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def addOneRow(self, root: Optional[TreeNode], val: int, depth: int) -> Optional[TreeNode]:
        if (depth == 1):
            return TreeNode(val, root)

        elif (depth == 2):
            root.left = TreeNode(val, root.left)
            root.right = TreeNode(val, None, root.right)
            return root

        else:
            if (root.left != None):
                root.left = self.addOneRow(root.left, val, depth-1)
            if (root.right != None):
                root.right = self.addOneRow(root.right, val, depth-1)
            return root

File: leetcode/add_one_row_to_tree/tree_new_row.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode addOneRow(TreeNode root, int val, int depth) {
	    if (depth == 1) {
		    return new TreeNode(val, root, null);
	    }

	    else if (depth == 2) {
		    root.left = new TreeNode(val, root.left, null);
		    root.right = new TreeNode(val, null, root.right);
		    return root;
	    }

	    else {
		    if (root.left != null) 
			    root.left = this.addOneRow(root.left, val, depth-1);
		    if (root.right != null) 
			    root.right = this.addOneRow(root.right, val, depth-1);
		    return root;
	    }
    }
}
